cvs diff: Diffing .
cvs diff: Diffing css
cvs diff: Diffing docs
cvs diff: Diffing handlers
cvs diff: Diffing help
cvs diff: Diffing help/images
cvs diff: Diffing images
cvs diff: Diffing includes
cvs diff: Diffing js
cvs diff: Diffing modules
cvs diff: Diffing modules/aggregator
cvs diff: Diffing modules/comment
cvs diff: Diffing modules/contact
cvs diff: Diffing modules/filter
cvs diff: Diffing modules/locale
cvs diff: Diffing modules/node
cvs diff: Diffing modules/profile
cvs diff: Diffing modules/search
cvs diff: Diffing modules/statistics
cvs diff: Diffing modules/system
cvs diff: Diffing modules/taxonomy
cvs diff: Diffing modules/translation
cvs diff: Diffing modules/upload
cvs diff: Diffing modules/user
cvs diff: Diffing plugins
cvs diff: Diffing theme
cvs diff: Diffing translations
cvs diff: Diffing views_export
? .buildpath
? .project
Index: handlers/views_handler_field.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/views/handlers/views_handler_field.inc,v
retrieving revision 1.30
diff -u -p -r1.30 views_handler_field.inc
--- handlers/views_handler_field.inc	21 Aug 2009 23:58:35 -0000	1.30
+++ handlers/views_handler_field.inc	31 Aug 2009 16:48:13 -0000
@@ -619,6 +619,9 @@ class views_handler_field extends views_
     if (isset($alter['fragment'])) {
       $options['fragment'] = strtr($alter['fragment'], $tokens);
     }
+    if (isset($this->options['alter']['language'])) {
+      $options['language'] = $this->options['alter']['language'];
+    }
 
     $value .= l($text, $path, $options);
 
Index: modules/node/views_handler_field_node.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/views/modules/node/views_handler_field_node.inc,v
retrieving revision 1.5
diff -u -p -r1.5 views_handler_field_node.inc
--- modules/node/views_handler_field_node.inc	21 Aug 2009 23:58:35 -0000	1.5
+++ modules/node/views_handler_field_node.inc	31 Aug 2009 16:48:13 -0000
@@ -15,6 +15,9 @@ class views_handler_field_node extends v
   function construct() {
     parent::construct();
     $this->additional_fields['nid'] = 'nid';
+    if (module_exists('translation')) {
+      $this->additional_fields['language'] = array('table' => 'node', 'field' => 'language');
+    }
   }
 
   function option_definition() {
@@ -45,6 +48,10 @@ class views_handler_field_node extends v
     if (!empty($this->options['link_to_node']) && $data !== NULL && $data !== '') {
       $this->options['alter']['make_link'] = TRUE;
       $this->options['alter']['path'] = "node/" . $values->{$this->aliases['nid']};
+      if (isset($this->aliases['language'])) {
+        $languages = language_list();
+        $this->options['alter']['language'] = $languages[$values->{$this->aliases['language']}]; 
+      }
     }
     return $data;
   }
